internal/rpc/jsonrpc/methods.go

```diff
@@ -2056,7 +2056,8 @@ func (s *Server) importPrivKey(ctx context.Context, icmd any) (any, error) {
 		s.wg.Add(1)
 		go func() {
 			defer s.wg.Done()
-			_ = w.RescanFromHeight(context.Background(), n, scanFrom)
+			serverCtx := s.httpServer.BaseContext(nil)
```

Comment:
We can use the outer context here instead of calling BaseContext ourselves.  It will be canceled when the process begins shutdown.

Reply:
or well no. let me test that as well. it's possible that this context is canceled after the handler returns....

Reply:
By outer context do you mean the one passed into `importPrivKey`? That will be cancelled as soon as a response is sent to the RPC client, hence why we want to use the server context instead.

Reply:
Yep, the rescan immediately exits with that suggested change.